avm-transpiler/src/transpile.rs

```diff
@@ -1060,23 +1061,39 @@ pub fn patch_debug_info_pcs(
     brillig_pcs_to_avm_pcs: &[usize],
 ) -> Vec<DebugInfo> {
     for patched_debug_info in debug_infos.iter_mut() {
-        // create a new map with all of its keys (OpcodeLocations) patched
-        let mut patched_locations: BTreeMap<OpcodeLocation, Vec<Location>> = BTreeMap::new();
-        for (original_opcode_location, source_locations) in patched_debug_info.locations.iter() {
-            match original_opcode_location {
-                OpcodeLocation::Brillig { acir_index, brillig_index } => {
-                    let avm_opcode_location = OpcodeLocation::Brillig {
-                        acir_index: *acir_index,
-                        // patch the PC
-                        brillig_index: brillig_pcs_to_avm_pcs[*brillig_index],
-                    };
-                    patched_locations.insert(avm_opcode_location, source_locations.clone());
+        assert_eq!(
+            patched_debug_info.brillig_locations.len(),
+            1,
+            "We should only be attempting to transpile a single Brillig function"
+        );
+        let mut patched_brillig_locations: BTreeMap<
+            BrilligFunctionId,
+            BTreeMap<OpcodeLocation, Vec<Location>>,
+        > = BTreeMap::new();
+        for (brillig_function_id, opcode_locations_map) in
+            patched_debug_info.brillig_locations.iter()
```

Comment:
This is the crux of the AVM changes in this PR @fcarreiro. All Brillig debug locations are going to be contained within a separate per function brillig locations map now.

Reply:
I could just fetch at id 0 here as we make the assumption already that we have a single brillig function. Let me know what you prefer.

Reply:
If it works I don't mind the solution! The transpiler is not a bottleneck now.